tests/components/powerwall/test_sensor.py

```diff
@@ -0,0 +1,116 @@
+    expected_attributes = {
+        "frequency": 60,
+        "energy_exported": 10429451.9916853,
+        "energy_imported": 4824191.60668611,
+        "instant_average_voltage": 120.650001525879,
+        "unit_of_measurement": "kWh",
+        "friendly_name": "Powerwall Site Now",
+        "device_class": "power",
+    }
+    # Only test for a subset of attributes in case
+    # HA changes the implementation and a new one appears
+    assert all(
+        item in entity.attributes.items() for item in expected_attributes.items()
+    )
+
+    entity = hass.states.get("sensor.powerwall_load_now")
+    assert entity.state == "1.971"
+    expected_attributes = {
+        "frequency": 60,
+        "energy_exported": 1056797.48917483,
+        "energy_imported": 4692987.91889705,
+        "instant_average_voltage": 120.650001525879,
+        "unit_of_measurement": "kWh",
+        "friendly_name": "Powerwall Load Now",
+        "device_class": "power",
+    }
+    # Only test for a subset of attributes in case
+    # HA changes the implementation and a new one appears
+    assert all(
+        item in entity.attributes.items() for item in expected_attributes.items()
+    )
+
+    entity = hass.states.get("sensor.powerwall_battery_now")
+    assert entity.state == "-8.55"
+    expected_attributes = {
+        "frequency": 60.014,
+        "energy_exported": 3620010,
+        "energy_imported": 4216170,
+        "instant_average_voltage": 240.56,
+        "unit_of_measurement": "kWh",
+        "friendly_name": "Powerwall Battery Now",
+        "device_class": "power",
+    }
+    # Only test for a subset of attributes in case
+    # HA changes the implementation and a new one appears
+    assert all(
+        item in entity.attributes.items() for item in expected_attributes.items()
+    )
+
+    entity = hass.states.get("sensor.powerwall_solar_now")
+    assert entity.state == "10.49"
+    expected_attributes = {
+        "frequency": 60,
+        "energy_exported": 9864205.82222448,
+        "energy_imported": 28177.5358355867,
+        "instant_average_voltage": 120.685001373291,
+        "unit_of_measurement": "kWh",
+        "friendly_name": "Powerwall Solar Now",
+        "device_class": "power",
+    }
+    # Only test for a subset of attributes in case
+    # HA changes the implementation and a new one appears
+    assert all(
+        item in entity.attributes.items() for item in expected_attributes.items()
+    )
+
+    entity = hass.states.get("sensor.powerwall_charge")
+    assert entity.state == "47.32"
+    expected_attributes = {
+        "unit_of_measurement": "%",
+        "friendly_name": "Powerwall Charge",
+        "device_class": "battery",
+    }
+    # Only test for a subset of attributes in case
+    # HA changes the implementation and a new one appears
+    assert all(
+        item in entity.attributes.items() for item in expected_attributes.items()
+    )
+
+    # "binary_sensor.grid_status"
```

Comment:
Is this a note to add more tests?

Reply:
I was making sure I check the state of each sensor and I didn't delete that one from my pasted list

Reply:
fixed